internal/serve/httphandler/payments_handler.go

```diff
@@ -125,7 +125,43 @@ func (p PaymentsHandler) RetryPayments(rw http.ResponseWriter, req *http.Request
 		return
 	}
 
-	err = p.Models.Payment.RetryFailedPayments(ctx, user.Email, reqBody.PaymentIDs...)
+	err = db.RunInTransaction(ctx, p.Models.DBConnectionPool, nil, func(dbTx db.DBTransaction) error {
```

Comment:
This is a good move. I'm assuming you put the transaction boundary at this level so that if the event fails to submit we rollback the db tx ?

Reply:
Exactly! In this ticket, I also wanted to improve our error handling when writing new messages.